pkg/controllers/leaderworkerset_controller.go

```diff
@@ -611,9 +611,10 @@ func rollingUpdatePartition(states []replicaState, stsReplicas int32, rollingSte
 	// (for example, all replicas are not ready when rolling update is started).
 	// Note that we never drop the partition below rolliingStepPartition.
 	for idx := min(partition, stsReplicas-1); idx >= rolliingStepPartition; idx-- {
-		if !states[idx].ready {
-			partition = idx
+		if states[idx].ready && states[idx].updated {
```

Comment:
Also involve @pierewoj in as the author of this.

Reply:
Spent a short while on this, I was able to reproduce this with an integ test. I agree the issue is that in this logic here we don't consider that it's actually OK to move the partition if the replica is already updated. But the proper condition seems to be:
```
if !states[idx].ready || states[idx].updated {
	partition = idx
} else {
	break
}
```

And with that the repro test (and other integ tests) succeed.

Reply:
Maybe we should be more strict here like `if !states[idx].ready || (states[idx].updated && states[idx].ready)`, what do your guys think? @pierewoj @gujingit It follows the same pattern in lws now as we try to make sure the updated ones should be ready first.

Reply:
true.
/lgtm
/approve

Thanks all!